### PR TITLE
test: use proceeds of predicate

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close https://github.com/FuelLabs/fuels-ts/issues/757

### CODE REVIEW

**Review by arboleya — DISMISSED** [review]

**Comment by arboleya** [comment]
I've read through the code, and it all makes sense, but I'm still unfamiliar with these areas.

I believe @camsjams's review will be decisive on this one.

**Review by Dhaiwat10 — DISMISSED** [review]

File: packages/fuel-gauge/src/predicate.test.ts [hunk 1]
```diff
@@ -469,4 +471,141 @@ describe('Predicate', () => {
...
+    // calling the script with the receiver account (no resources)
+    const scriptInput = 1;
+    scriptInstance.account = receiver;
+    await expect(scriptInstance.functions.main(scriptInput).call()).rejects.toThrow(
+      /not enough resources to fit the target/
+    );
+
+    // setup predicate
+    const amountToPredicate = 100;
+    const amountToReceiver = 50;
```
[the remark this answers is not in the document]
Comment (Torres-ssf) [reply packages/fuel-gauge/src/predicate.test.ts:505]: It was left in the Predicate, I guess :thinking: . I've adjusted the value to ensure only the necessary amount is passed to the Predicate

**Comment by Torres-ssf** [comment]
@arboleya @Dhaiwat10 @camsjams I've just made a minor refact in the test code. Nothing changed

**Review by arboleya — DISMISSED** [review]
I see the change, but it doesn't make any difference - does it? Perhaps what @camsjams meant is that you could try to validate the remaining coins somehow. Maybe I got it wrong.

File: packages/fuel-gauge/src/predicate.test.ts [hunk 2]
```diff
@@ -601,11 +601,16 @@ describe('Predicate', () => {
...
+    const remainingPredicateBalance = await predicate.getBalance();
 
     expect(toNumber(initialReceiverBalance)).toBe(0);
 
     expect(bn(initialReceiverBalance).add(amountToReceiver).toNumber()).toEqual(
       bn(finalReceiverBalance).add(contractAmount).add(gasPrice).toNumber()
     );
+
+    expect(toNumber(remainingPredicateBalance)).toBeGreaterThanOrEqual(
+      bn(amountToPredicate).add(amountToReceiver).toNumber()
```
[the remark this answers is not in the document]
Comment (Torres-ssf) [reply packages/fuel-gauge/src/predicate.test.ts]: @camsjams The same Predicate is being widely used in this test file. Many test cases are adding assets to it and altering its balance. I can't tell what will be the right amount at the time of execution. 

But I have fixed both test suites to properly assure that only the expected values are being used.

Please let me know if I am missing something.

**Review by Dhaiwat10 — APPROVED** [review]

**Comment by Dhaiwat10** [comment]
@Torres-ssf there are some merge conflicts that might need solving!

**Review by arboleya — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Comment by Torres-ssf** [comment]
> @Torres-ssf there are some merge conflicts that might need solving!

@Dhaiwat10 Solved already.

**Pull request merged**